src/algorithms/mts.jl

```diff
@@ -0,0 +1,354 @@
+
+# MTS Algorithms
+# MTS: Multiple Trajectory Search for Large Scale Global Optimization, 
+# By [Lin-Yu Tseng and Chun Chen, 2008](https://sci2s.ugr.es/sites/default/files/files/TematicWebSites/EAMHCO/contributionsCEC08/tseng08mts.pdf)
+
+using Random: randperm
+
+@with_kw struct MTSOptions
+    method::Function = MTS
+    M = 100
+    maxiter=200
+    search_range_tol=1e-15
+    n_foreground = 80
+    n_local_search = 200
+    n_local_search_test = 3
+    n_local_search_best = 300
+    BONUS1 = 10
+    BONUS2 = 1 
+    a_min = 0.4
+    a_max = 0.5
+    b_min = 0.1
+    b_max = 0.3
+    c_min = 0
+    c_max = 1
+    function MTSOptions(args...; kwargs...)
+        @assert (length(args) > 0 && args[1] in _MTS_OPTIMIZATION_METHODS) || kwargs[:method] in _MTS_OPTIMIZATION_METHODS
+        new(args...; kwargs...)
+    end
+end
+
+@params mutable struct MTSWorkspace <: Workspace
+    model::VecModel
+    x0::AbstractVector
+    x::AbstractVector
+    options::MTSOptions
+    enable::AbstractVector
+    improve::AbstractVector
```

Comment:
if the vector is a vector of boolean, you can use `BitVector`. It can be initialized with `trues(n)` for a vector of n elements starting with only `true` or `falses(n)`